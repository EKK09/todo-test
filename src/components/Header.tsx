import React from 'react';
import styled from 'styled-components';
import { useTranslation } from 'react-i18next';
import useUserStore from 'src/store/user';
import LangSelector from './LangSelector';

const Wrapper = styled.div`
  padding: 26px 0;
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const Title = styled.div`
  font-size: 2rem;
  text-align: center;
  color: palevioletred;
  line-height: 26px;
`;

const OperationWrapper = styled.div`
  position: absolute;
  display: flex;
  flex-wrap: nowrap;
  align-items: center;
  right: 0;
`;

const Button = styled.button`
  color: black;
  font-weight: bolder;
  background-color: white;
  border-radius: 4px;
  cursor: pointer;
  float: right;
`;

const Header = (): React.ReactElement => {
  const { t } = useTranslation();
  const { id, setId } = useUserStore();
  const handleLogout = (): void => {
    setId(null);
  };
  const isShowLogoutBtn: boolean = id !== null;
  return (
    <Wrapper>
      <Title>{t('global.appName')}</Title>
      <OperationWrapper>
        <LangSelector />
        {isShowLogoutBtn ? <Button type="button" onClick={handleLogout}>{t('global.logout')}</Button> : null}
      </OperationWrapper>
    </Wrapper>
  );
};

export default Header;
