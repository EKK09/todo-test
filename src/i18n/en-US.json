{
    "global": {
        "appName": "TODO LIST",
        "logout": "Logout"
    },
    "loginPage": {
        "account": "Account",
        "password": "Password",
        "login": "Login"
    },
    "todo": {
        "completed": "Completed"
    }
}