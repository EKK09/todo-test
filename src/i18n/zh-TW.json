{
    "global": {
        "appName": "代辦事項"
    },
    "loginPage": {
        "account": "帳號",
        "password": "密碼",
        "login": "登入"
    },
    "todo": {
        "completed": "完成"
    }
}