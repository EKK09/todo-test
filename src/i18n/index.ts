import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import enUSResource from './en-US.json';
import zhTWResource from './zh-TW.json';

export enum Lang {
    EN_US = 'en-US',
    ZH_TW = 'zh-TW',
}

export const LANG_OPTION = {
  [Lang.EN_US]: 'English',
  [Lang.ZH_TW]: '中文',
};

const resources = {
  [Lang.EN_US]: {
    translation: enUSResource,
  },
  [Lang.ZH_TW]: {
    translation: zhTWResource,
  },
};

const getDefaultLang = (locale: string): Lang => {
  switch (locale) {
    case Lang.EN_US:
      return Lang.EN_US;

    case Lang.ZH_TW:
      return Lang.ZH_TW;

    default:
      return Lang.EN_US;
  }
};

const locale = window.navigator.language;

i18n
  .use(initReactI18next)
  .init({
    resources,
    lng: getDefaultLang(locale),
  });

export default i18n;
