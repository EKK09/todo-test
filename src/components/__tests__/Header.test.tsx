import React from 'react';
import { render } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import i18n, { Lang } from 'src/i18n';
import Header from '../Header';

jest.mock('src/components/LangSelector', () => 'Mock-LangSelector');

describe('Header component', () => {
  it('Render Header english version', async () => {
    i18n.changeLanguage(Lang.EN_US);
    const wrapper = render(
      <I18nextProvider i18n={i18n}>
        <Header />
      </I18nextProvider>,
    );

    expect(wrapper.container).toMatchSnapshot();
  });

  it('Render Header chinese version', async () => {
    i18n.changeLanguage(Lang.ZH_TW);
    const wrapper = render(
      <I18nextProvider i18n={i18n}>
        <Header />
      </I18nextProvider>,
    );

    expect(wrapper.container).toMatchSnapshot();
  });
});
