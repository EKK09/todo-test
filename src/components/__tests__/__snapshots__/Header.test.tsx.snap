// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Render Header chinese version 1`] = `
<div>
  <div
    class="sc-bdfBQB fOxfaE"
  >
    <div
      class="sc-gsTEea jdCHeA"
    >
      待辦事項
    </div>
    <div
      class="sc-dlfnuX knTXeM"
    >
      <mock-langselector />
    </div>
  </div>
</div>
`;

exports[`Header component Render Header english version 1`] = `
<div>
  <div
    class="sc-bdfBQB fOxfaE"
  >
    <div
      class="sc-gsTEea jdCHeA"
    >
      TODO LIST
    </div>
    <div
      class="sc-dlfnuX knTXeM"
    >
      <mock-langselector />
    </div>
  </div>
</div>
`;
